### PR TITLE
chore(logging): Add support to "off" for logging level

### LINKED ISSUE #3051: Can not turn off logging
### Environment
- latest `getsentry/relay` image
- Run on GKE Kubernetes

### Steps to reproduce
1. Run sentry with the following configuration
```
relay:
  mode: static
  upstream: https://<organization-domain>
  host: 0.0.0.0
  port: 3000
logging:
  level: off
```

### Expected Result
Stopping the log as described in https://docs.sentry.io/product/relay/options/#logging

### Actual Result
```
caused by: logging.level: invalid value: string "off", expected a log level at line 7 column 10
```
The relay works fine when I use `info` or `error` as the log level value, but it gives me that error when I use `off` log level value

### PR DESCRIPTION
**Pull request by olksdr**
Support all the logging levels mentioned in the docs https://docs.sentry.io/product/relay/options/#logging

This change basically addresses the support for `"off"` level.

fix: https://github.com/getsentry/relay/issues/3051

#skip-changelog

### CODE REVIEW

File: relay-log/src/setup.rs [hunk 1]
```diff
@@ -47,53 +49,89 @@ pub enum LogFormat {
...
-            fn visit_str<E>(self, value: &str) -> Result<Level, E>
-            where
-                E: Error,
-            {
-                value
-                    .parse()
-                    .map_err(|_| Error::invalid_value(Unexpected::Str(value), &self))
-            }
+    /// Returns the tracing logging [`tracing::Level`].
+    pub fn level(&self) -> TracingLevel {
```
Comment (Dav1dde) [inline relay-log/src/setup.rs]: Should we return `Option<TracingLevel>` here instead?

File: relay-log/src/setup.rs [hunk 2]
```diff
@@ -47,53 +49,89 @@ pub enum LogFormat {
...
-    pub fn deserialize<'de, D>(deserializer: D) -> Result<Level, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        struct V;
-
-        impl<'de> Visitor<'de> for V {
-            type Value = Level;
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
```
Comment (Dav1dde) [inline relay-log/src/setup.rs]: We have `relay_common::impl_str_serde` to make sure it uses the same impl as `FromStr`. E.g. the `FromStr` allows mixed case, this serde derive does not.

**Review by Dav1dde — APPROVED** [review]

File: relay-log/src/setup.rs (hunk 1 above)
Comment (olksdr) [reply relay-log/src/setup.rs]: removed this func

File: relay-log/src/setup.rs (hunk 2 above)
Comment (olksdr) [reply relay-log/src/setup.rs]: using `impl_str_serde` now

File: relay-log/src/setup.rs [hunk 3]
```diff
@@ -47,53 +50,78 @@ pub enum LogFormat {
...
-        struct V;
-
-        impl<'de> Visitor<'de> for V {
-            type Value = Level;
-
-            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
-                formatter.write_str("a log level")
-            }
+#[derive(Clone, Copy, Debug)]
+pub enum Level {
```
Comment (iker-barriocanal) [inline relay-log/src/setup.rs:68]: nit(?)
```suggestion
pub enum LogLevel {
```

**Review by iker-barriocanal — APPROVED** [review]

File: relay-log/src/setup.rs (hunk 3 above)
Comment (jjbayer) [reply relay-log/src/setup.rs:68]: [Avoid redundant prefixes](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#avoid-redundant-prefixes-%5Brfc-356%5D).

**Pull request merged**